### DISCUSSION TITLE
Using InitialPeers with ROS2 over WiFi

### DISCUSSION THREAD

**Discussion by cocharland** [eProsima/Fast-DDS#3420] [Q&A]
Hello!
I am trying to reduce the amount of multicast traffic being sent over wifi. My endpoint doesn't have the ability to set the discovery server so I can't just use that easily. My setup is a rover with a router strapped to it and an android based controller. The multicast traffic from the rover is swamping the network.  I've tried the following setup for the xml but services/topics on the robot are failing to connect locally. I am unable to see nodes/topics from ros2 topic list at all from the rover to itself.
```
<?xml version="1.0" encoding="UTF-8" ?>
<profiles xmlns="http://www.eprosima.com/XMLSchemas/fastRTPS_Profiles">
    <participant profile_name="default" is_default_profile="true">
        <rtps>
            <builtin>
                <metatrafficUnicastLocatorList>
                    <locator/>
                </metatrafficUnicastLocatorList>
                <initialPeersList>
                    <locator>
                        <udpv4>
                            <address>192.168.1.2</address>
                            <address>192.168.1.10</address>
                        </udpv4>
                    </locator>
                </initialPeersList>
            </builtin>
        </rtps>
    </participant>
</profiles>
```
Any thoughts? The maxInitialPeersRange didn't seem to help the system either.

**Comment by MiguelCompany** [comment]
@cocharland You need one locator per address, so

```xml
<?xml version="1.0" encoding="UTF-8" ?>
<profiles xmlns="http://www.eprosima.com/XMLSchemas/fastRTPS_Profiles">
    <participant profile_name="default" is_default_profile="true">
        <rtps>
            <builtin>
                <metatrafficUnicastLocatorList>
                    <locator/>
                </metatrafficUnicastLocatorList>
                <initialPeersList>
                    <locator>
                        <udpv4>
                             <address>192.168.1.2</address>
                        </udpv4>
                    </locator>
                    <locator>
                        <udpv4>
                            <address>192.168.1.10</address>
                        </udpv4>
                    </locator>
                </initialPeersList>
            </builtin>
        </rtps>
    </participant>
</profiles>
```

**Reply by cocharland** [reply]
This has allowed my nodes to discover eachother again. Thanks. I still am unable to achieve acceptable performance over wifi. Do you have any suggestions on ways to improve wifi performance with fastDDS? It seems like the docs all point to the discovery server, but I don't think my endpoint(ros2-android) really supports that. I'll keep looking
